projects/plugins/jetpack/_inc/client/newsletter/newsletter-categories.jsx

```diff
@@ -0,0 +1,86 @@
+import { ToggleControl } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import SettingsCard from 'components/settings-card';
+import SettingsGroup from 'components/settings-group';
+import React, { useCallback, useState } from 'react';
+import { connect } from 'react-redux';
+import { withModuleSettingsFormHelpers } from '../components/module-settings/with-module-settings-form-helpers';
+import TextInput from '../components/text-input';
+import Textarea from '../components/textarea';
+
+const mapCategoriesIds = category => {
+	switch ( typeof category ) {
+		case 'number':
+			return category;
+		case 'string':
+			return parseInt( category );
+		case 'object':
+			return category.term_id;
+	}
+};
+
+/**
+ * NewsletterCategories settings component.
+ *
+ * @param {object} props - Component props.
+ * @returns {React.Component} Subscription settings component.
+ */
+function NewsletterCategories( props ) {
+	const {
+		updateFormStateModuleOption,
+		isNewsletterCategoriesEnabled,
+		newsletterCategories,
+		updateFormStateOptionValue,
+		categories,
+	} = props;
+
+	const [ newCategories, setNewCategories ] = useState( '' );
+
+	const handleEnagleNewsletterCategoriesToggleChange = useCallback( () => {
+		updateFormStateModuleOption( 'subscriptions', 'wpcom_newsletter_categories_enabled' );
+	}, [ updateFormStateModuleOption ] );
+
+	const categoriesValue = JSON.stringify( newsletterCategories.map( mapCategoriesIds ) );
+	const parseArray = useCallback( () => {
+		try {
+			updateFormStateOptionValue( 'wpcom_newsletter_categories', JSON.parse( newCategories ) );
+		} catch ( error ) {
+			alert( 'Invalid JSON' );
+		}
+	}, [ newCategories, updateFormStateOptionValue ] );
+
+	const onCategoriesChange = useCallback( e => {
+		setNewCategories( e.target.value );
+	}, [] );
+
+	return (
+		<SettingsCard { ...props } module="subscriptions">
+			<SettingsGroup hasChild disableInOfflineMode disableInSiteConnectionMode>
```

Comment:
Why `disableInSiteConnectionMode`? 🤔

Reply:
Looks like we do need it. Apparently it is possible to connect a self hosted site to Jetpack without setting up WPCOM account (p1677772292841239-slack-CBG1CP4EN). When that happens we do not say that the site is WPCOM connected, but instead we call it site connected.